### PR TITLE
Add captions to gallery block

### LINKED ISSUE #1443: Gallery: add support for optional captions
Individual images should have optional captions.

### PR DESCRIPTION
**Pull request by mtias**
image

Closes #1443.

### CODE REVIEW

File: blocks/library/gallery/style.scss [hunk 1]
```diff
@@ -19,3 +19,13 @@
...
 	margin: 1.8em 0;
 }
+
+.wp-block-gallery .blocks-editable {
+	position: absolute;
+	bottom: 0;
+	left: 0;
+	right: 0;
+	figcaption {
+		position: static;
```
Comment (iseulde) [inline blocks/library/gallery/style.scss]: Why is this needed? I believe ice will re-add it as inline style.

Comment (mtias) [inline blocks/library/gallery/style.scss]: The positioning is a bit messed up because we'd want to apply it to figcaption on the front end, but to editable in the editor.

**Comment by mkaz** [comment]
I can use some clarification for user experience, the core Media Modal which is used for selecting the image also includes ordering images, and editing captions as part of the standard create gallery flow. This is improved on create with PR #1820 

Do we want those interactions to happen within Gutenberg block also, or alternatively could remove from modals (might require core-media work), or just manage within the modals?

cc: @jasmussen and since he's on holiday, @melchoyce

**Comment by jasmussen** [comment]
> I can use some clarification for user experience, the core Media Modal which is used for selecting the image also includes ordering images, and editing captions as part of the standard create gallery flow. This is improved on create with PR #1820

> Do we want those interactions to happen within Gutenberg block also, or alternatively could remove from modals (might require core-media work), or just manage within the modals?

In general, the more interactions that can happen _on the block itself_ (without adding undue complexity), the better. The idea is that you should be able to get fully going with a block, with using only the block interface itself. Any advanced configuration can happen in the sidebar/inspector. See also https://github.com/WordPress/gutenberg/blob/master/docs/design.md#block-interface--sidebar--inspector

In the case of the gallery, I see the media modal eventually going away entirely. That is, everything we want to port from the media modal, we should, including sort order. But this doesn't have to happen immediately, and the idea is to move towards "the block" being the only interface you ever have to learn.

File: blocks/library/gallery/gallery-image.js [hunk 2]
```diff
@@ -1,8 +1,25 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from 'i18n';
```
Comment (aduth) [inline blocks/library/gallery/gallery-image.js]: Note with #2172, these need to be updated to prefix the dependencies with `@wordpress/`. You will need to perform a rebase against the latest version of master and apply your changes:

```
git fetch origin
git rebase origin/master
```

**Comment by youknowriad** [comment]
Looks like this has been superseded by https://github.com/WordPress/gutenberg/pull/4199
